### DISCUSSION TITLE
What is the scope of changelog group regex?

### DISCUSSION THREAD

**Discussion by calebstewart** [goreleaser/goreleaser#4259] [Q&A] [answered]
The documentation is a bit vague, and I wanted to clarify the scope of the regex for groups within the changelog.

Are the regular expressions executed over the entire commit message or over just the title/first line of the commit message that is displayed in the changelog? For example, if the commit message was:

```
Added some new feature

feat: something else
```

Would that match the example regex `^.*?feat(\([[:word:]]+\))??!?:.+$`? The intent in that case would be that the friendlier "Added some new feature" message is displayed in the changelog, but it is grouped and/or filtered by the less-friendly "feat: something else" part in the commit message body.

This might not be a common thing, and if the regex only matches the title/first line, then that's totally fine. I just wanted clarification so I know how to write things, and what to expect. :)

Given that there's no way to generate the changelog locally without the Pro version, I have no way to test without actually creating a release, which I'd prefer not to do. Otherwise, I would have just guessed and checked locally. :P

**Comment by calebstewart** [comment, marked as answer]
Looking at the implementation in [changelog.go], it seems that the regular expressions are executed on only the commit message as viewed in the changelog (i.e. the first line/title). That's understandable in this context, but I just wanted to make sure. I  will wait to mark this as the answer to my question until validated by someone who has looked at the code for more than 60 seconds, though.

It may be worthwhile to add a note of that in the documentation for the changelog configuration, though.

[changelog.go]: https://github.com/goreleaser/goreleaser/blob/main/internal/pipe/changelog/changelog.go#L248

**Reply by caarlos0** [reply]
yes, you're correct, its the first line only, as the output of `git log --pretty=oneline`

**Reply by caarlos0** [reply]
also just updated the docs in 76f2b81bb503277d5c63b71e78a8e50637bb86b7
